src/celeritas/track/SimTrackView.hh

```diff
@@ -95,11 +95,14 @@ class SimTrackView
     // Whether the track is alive or inactive or dying
     CELER_FORCEINLINE_FUNCTION TrackStatus status() const;
 
-    // Limiting step and action to take
-    CELER_FORCEINLINE_FUNCTION StepLimit const& step_limit() const;
+    // Limiting step
+    CELER_FORCEINLINE_FUNCTION real_type step_length() const;
 
-    // Mutable access to step limit (TODO: hack)
-    CELER_FORCEINLINE_FUNCTION StepLimit& step_limit();
+    // Mutable access to limiting step
+    CELER_FORCEINLINE_FUNCTION real_type& step_length();
```

Comment:
we probably want to have a consistent way of updating the step length, along_step_action and post_step_action. Currently, we return a reference to step length and along_step_action and we have a setter force_step_limit for the post_step_action.

Reply:
Yeah. How about a `void step_limit(real_type length, ActionId post_step);` and a `void step_limit(ActionId post_step);` for the times where we directly set the post-step action? Maybe we should take a quick survey of when these are used?

Reply:
There's one case where the step length is update alone in [UrbanMsc](https://github.com/celeritas-project/celeritas/blob/e16e9285edecaeb619c2f614c4deb16864050fce/src/celeritas/em/msc/UrbanMsc.hh#L182-L187). `step_limit(real_type length, ActionId post_step)` checks that the step length is limiting before updating the action and the step length, we sometimes need to update these regardless (e.g. looping track in PropagationApplier)

Reply:
OK. Let's just add raw accessors `void step_length(real_type length)` and `void post_step_action(ActionId post_step)`; and the first one can at least assert that the length is strictly positive.